load-balance/src/LoadBalanceStep.tsx

```diff
@@ -0,0 +1,178 @@
+                            lb.configuration.set(e[0], tmp);
+                        }
+                    });
+                    setLbConfiguration(new Map(lb.configuration));
+                    setCurrentLoadBalancer(lb);
+                }
+            }
+        }, [step.branches, step.parameters])
+
+        const onMinus = () => {
+            const newValue = (numberOfBranches || 0) - 1;
+            setNumberOfBranches(newValue);
+            handleDeleteBranch();
+        };
+
+        const onChange = (event: React.FormEvent<HTMLInputElement>) => {
+            const value = (event.target as HTMLInputElement).value;
+            setNumberOfBranches(+value);
+        };
+
+        const onPlus = () => {
+            const newValue = (numberOfBranches || 0) + 1;
+            setNumberOfBranches(newValue);
+            addBranch();
+        };
+
+        function updateStepParams() {
+            const configParams = {};
+            lbConfiguration.forEach((v, k) => {
+                // @ts-ignore
+                configParams[k] = v.value;
+            });
+
+            const p = {[currentLoadBalancer.id]: configParams}
+            const params = step.parameters;
+            params[0].value = p;
+
+            step.parameters = params;
+            props.updateStep(step)
+            props.notifyKaoto(`Load Balance Configuration updated`);
+        }
+
+        function handleDeleteBranch() {
+            const index = numberOfBranches - 1;
+            if (step.branches && index && index !== -1) {
+                step.branches = step.branches.filter((_b, i) => i !== index);
+                setStep(step);
+                props.notifyKaoto(`Endpoint deleted`);
+                props.updateStep(step);
+            }
+        }
+
+        function addBranch() {
+            const branch = {
+                identifier: '',
+                steps: []
+            }
+            let index = -1;
+            if (!step.branches) {
+                step.branches = [branch];
+                index = 0;
+            } else {
+                index = step.branches?.length;
+                step.branches.splice(index, 0, branch);
+            }
+            setStep(step);
+            props.notifyKaoto(`Branch added`);
+            props.updateStep(step);
+        }
+
+        function onChangeLoadbalancerType(value: string) {
+            const found = loadBalanceStrategies.find(lb => lb.name === value);
+            if (found) {
+                setLbConfiguration(found.configuration);
+                setCurrentLoadBalancer(found);
+            }
+        }
+
+        const updateParameterValue = (key: string, newValue: string | number | boolean) => {
+            const tmpField: any = {...lbConfiguration.get(key), value: newValue};
+            const tmpProperties: Map<string, ILBConfigurationField> = lbConfiguration
+            tmpProperties.set(key, tmpField);
+            setLbConfiguration(new Map(tmpProperties));
```

Comment:
Is it needed to create a new `Map` in every method call?

Reply:
technically yes, afaik reference would stay the same and the fields wouldn't re-render with new value.